### PR TITLE
Clean up gn flags for the develop/profile/release runtime modes

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

File: sky/tools/gn [hunk 1]
```diff
@@ -126,13 +120,10 @@ def parse_args(args):
   args = args[1:]
   parser = argparse.ArgumentParser(description='A script run` gn gen`.')
 
-  parser.add_argument('--debug', default=True, action='store_true')
-  parser.add_argument('--release', default=False, dest='debug', action='store_false')
+  parser.add_argument('--debug', default=False, action='store_true')
```
Comment (abarth) [inline sky/tools/gn]: It's probably worth an email to the team about changing the default from Debug to Release and about renaming the out directories.

File: sky/tools/gn [hunk 2]
```diff
@@ -126,13 +120,10 @@ def parse_args(args):
...
+
+  parser.add_argument('--runtime-mode', type=str, choices=['develop', 'profile', 'release'], default='develop')
```
Comment (abarth) [inline sky/tools/gn]: These names don't quite line up with these names: https://github.com/flutter/flutter/wiki/Flutter's-modes

We should make them consistent, either by changing the wiki or changing the names here.  @Hixie

**Comment by abarth** [comment]
LGTM once the names are consistent in the code an in the wiki.

File: sky/tools/gn (hunk 2 above)
Comment (jason-simmons) [inline sky/tools/gn]: develop/profile/release matches the names used in Dart's runtime modes.
I also prefer keeping the "develop" runtime mode distinct from the "debug" build configuration.

I'll update the wiki.

**Comment by jason-simmons** [comment]
Also updated the path in the travis build script

File: sky/tools/gn (hunk 2 above)
Comment (Hixie) [inline sky/tools/gn]: I think matching what everyone in the industry uses ("debug" and "release") is much, much more valuable than matching what Dart uses. (Search google for [debug build] and [develop build].)

I agree that there's some confusion between `flutter --debug` and `gn --debug`. I think we can mitigate this largely by referring to the latter one consistently as "engine-debug". I think that confusion will be _much_ less than the confusion that "develop" would introduce.

**Pull request merged**